### PR TITLE
VL - Proposed Models

### PR DESCRIPTION
**Pull request by VioletSnow08**

### CODE REVIEW

File: ADSBackend/Models/Member.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        public string PhoneNumber { get; set; }
+
+        [Display(Name = "Profile Picture")]
+        public string profileImageSource { get; set; }
+
+        [Display(Name = "Bio")]
+        public string Description { get; set; }
+
+        [Required]
```
Comment (tanczosm) [inline ADSBackend/Models/Member.cs:44]: Add [JsonIgnore] to Password and Password Salt

File: ADSBackend/Models/Comment.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace ADSBackend.Models
+{
+	public class Comment
+	{
+		[Key]
+		public int CommentId { get; set; }
+		public int AuthorId { get; set; }
+		public Member Author { get; set; }
+		public int ParentCommentId { get; set; }
```
Comment (tanczosm) [inline ADSBackend/Models/Comment.cs:15]: add: public Comment Parent {get; set; }

File: ADSBackend/Models/ChatMessage.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ADSBackend.Models
+{
+	public class ChatMessage
+	{
+		public int MessageId { get; set; }
+		public Member Author { get; set; }
```
Comment (tanczosm) [inline ADSBackend/Models/ChatMessage.cs:12]: add public int AuthorId {get; set;}

File: ADSBackend/Models/ChatMessage.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+		public Member To { get; set; }
```
Comment (tanczosm) [inline ADSBackend/Models/ChatMessage.cs:14]: add public int ToId {get; set; }

File: ADSBackend/Models/ChatMessage.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+		public string Body { get; set; }
+		public string Image { get; set; }
+		public DateTime CreatedAt { get; set; }
+		public DateTime EditedAt { get; set; }
+		public bool isDeleted { get; set; } = false;
```
Comment (tanczosm) [inline ADSBackend/Models/ChatMessage.cs]: Capitalize IsDeleted

File: ADSBackend/Models/ChatSession.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+namespace ADSBackend.Models
+{
+	public class ChatSession
+	{
+		public int SessionId { get; set; }
+		public Session Session { get; set; }
+		public Member Member1 { get; set; }
+		public Member Member2 { get; set; }
+		public List<ChatMessage> Messages { get; set; }
+		public DateTime LastMessageSentAt { get; set; }
```
Comment (tanczosm) [inline ADSBackend/Models/ChatSession.cs]: Add [NotMapped] 
public Message LastMessage { get; set; }

File: ADSBackend/Models/ChatSession.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ADSBackend.Models
+{
+	public class ChatSession
+	{
+		public int SessionId { get; set; }
+		public Session Session { get; set; }
+		public Member Member1 { get; set; }
```
Comment (tanczosm) [inline ADSBackend/Models/ChatSession.cs]: Remove Member1, Member2, etc.
Add public List<ChatMember> ChatMembers {get; set;}

File: ADSBackend/Models/Post.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+		public int PostId { get; set; }
+		public int AuthorId { get; set; }
+		public Member Author { get; set; }
+		public string Title { get; set; }
+		public string Body { get; set; }
+		public string Image { get; set; }
+		public bool isMachinePost { get; set; } = false;
+		public DateTime CreatedAt { get; set; }
+		public DateTime EditedAt { get; set; }
+		public bool isDeleted { get; set; } = false;
```
Comment (tanczosm) [inline ADSBackend/Models/Post.cs]: Rename isDeleted to IsDeleted

File: ADSBackend/Models/Post.cs [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+		public List<PostReaction> Reactions { get; set; }
+		public List<PostComment> Comments { get; set; }
+
```
Comment (tanczosm) [inline ADSBackend/Models/Post.cs:29]: Add
public int FavoriteCount { get; set; }
public bool IsFeatured { get; set; }
[NotMapped]
public bool IsFavorite { get; set; }

**Review by tanczosm — CHANGES_REQUESTED** [review]
Fix these issues and I will review again.  Click "Resolve Conversation" when you have completed each issue on the comment that goes with the issue.

File: ADSBackend/Models/Comment.cs (hunk 2 above)
Comment (VioletSnow08) [reply ADSBackend/Models/Comment.cs:15]: Would we need to have a ParentId also?(The including an ID and model is new to me sorry)

File: ADSBackend/Models/Post.cs (hunk 8 above)
Comment (VioletSnow08) [reply ADSBackend/Models/Post.cs]: Also changed IsMachinePost

File: ADSBackend/Models/Comment.cs (hunk 2 above)
Comment (tanczosm) [reply ADSBackend/Models/Comment.cs:15]: It's there as ParentCommentId

File: ADSBackend/Models/ChatSession.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ADSBackend.Models
+{
+	public class ChatSession
+	{
+		public int SessionId { get; set; }
```
Comment (tanczosm) [inline ADSBackend/Models/ChatSession.cs]: Rename SessionId to ChatSessionId

File: ADSBackend/Models/ChatSession.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+		public Session Session { get; set; }
```
Comment (tanczosm) [inline ADSBackend/Models/ChatSession.cs]: Get rid of the Session line

File: ADSBackend/Models/ChatSession.cs [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+		public List<Member> ChatMembers { get; set; }
```
Comment (tanczosm) [inline ADSBackend/Models/ChatSession.cs]: Unfortunately you can't do this.  Nowhere in the database will it store the list of members that are part of the chat session.  A single solitary member does not have any attachment in the Member class to a particular ChatSession so you're going to need another class to act as the glue that tracks which members belong to which chat sessions. 
Instead, create a new class like this:

ChatSessionMember
{
public int ChatSessionId {get; set;}
public ChatSession ChatSession {get; set;}
public int MemberId {get; set;}
public Member Member {get; set;}
}

And then change this line to:
public List<ChatSessionMember> ChatMembers {get; set;}

**Review by tanczosm — CHANGES_REQUESTED** [review]
Looks better.  Just a few more issues to address before we can merge.

File: ADSBackend/Models/Comment.cs (hunk 2 above)
Comment (VioletSnow08) [reply ADSBackend/Models/Comment.cs:15]: Oh my gosh I apologize 🤦 ~ it's been a long day for me. I apologize about that let me submit the commit and finish that up!

**Pull request merged**